### PR TITLE
Attempt at fixing flaky E2E tests

### PR DESCRIPTION
**Pull request by jahow**
* Download file on click: the targeted file is now mocked so we won't be relying on an actual third-party endpoint
* WMS query params order was changed since the update of OpenLayers in #698, this PR fixes the interception pattern
* Sorting by date changes the results order: this test was failing due to a race condition, fixed in this PR
* Improved the login command to avoid race conditions (now everything coming after it will be chained instead of run in parallel)

I also simplified the JSON fixture which was quite big (hence the big diff)

### CODE REVIEW

File: apps/datahub-e2e/src/e2e/datasets.cy.ts [hunk 1]
```diff
@@ -2,7 +2,6 @@ import 'cypress-real-events'
 
 describe('datasets', () => {
   beforeEach(() => {
-    cy.clearCookies()
```
Comment (tkohr) [inline apps/datahub-e2e/src/e2e/datasets.cy.ts:5]: Where are the cookies cleaned now?

File: apps/datahub-e2e/src/e2e/datasets.cy.ts [hunk 2]
```diff
@@ -445,15 +444,15 @@ describe('datasets', () => {
...
-            .should('not.eql', initialResultTitles)
-        })
+      it('changes the results order', function () {
+        cy.get('@results')
+          .find('[data-cy="recordTitle"]')
+          .then(($titles) => {
+            const titles = $titles
+              .toArray()
+              .map((title) => title.innerText.trim())
+            assert.notEqual(titles, this.initialResultTitles) // @initialResultTitles
```
Comment (tkohr) [inline apps/datahub-e2e/src/e2e/datasets.cy.ts:454]: Why do we need to access `initialResultTitles` in the `this` scope here? I guess `.should('not.eql', @initialResultTitles)` is not the same?

**Review by tkohr — APPROVED** [review]
Thanks a lot for the various tweaks @jahow!

File: apps/datahub-e2e/src/e2e/datasets.cy.ts (hunk 1 above)
Comment (jahow) [reply apps/datahub-e2e/src/e2e/datasets.cy.ts:5]: cypress clears cookies between each run by default: https://docs.cypress.io/api/commands/clearcookies

File: apps/datahub-e2e/src/e2e/datasets.cy.ts (hunk 2 above)
Comment (jahow) [reply apps/datahub-e2e/src/e2e/datasets.cy.ts:454]: This is a subtlety of cypress: doing `cy.get('@myAlias')` will recompute the chain for the alias, whereas doing `this.myAlias` will return the value that was computed earlier. See:
https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Avoiding-the-use-of-this

Comment (tkohr) [reply apps/datahub-e2e/src/e2e/datasets.cy.ts:454]: Ah, thanks for the explanations :-)

**Pull request merged**